src/app/common/search-select/search-select.component.ts

```diff
@@ -9,28 +9,60 @@ import { SkillService } from 'src/app/skill.service';
   styleUrls: ['./search-select.component.scss']
 })
 export class SearchSelectComponent implements OnInit {
-  constructor(private skillService: SkillService) {}
+  constructor(private skillService: SkillService,
+    private eRef: ElementRef) {
+    this.allSkills = [];
+    this.skillExistsEmitter.emit(this.skillExists);
```

Comment:
Why are you emitting skillExists in the constructor? I believe you only need to emit when the value has been modified, but in the constructor, nothing has happened yet that would have been able to modify the value.